akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1435,6 +1435,7 @@ private[stream] object Collect {
     private var groupEmitted = true
     private var finished = false
     private var totalWeight = 0L
+    private var elementCount = 0L
```

Comment:
Hmm, yes you are right. I think it already feels like a lot of a change for a minimal use case, but maybe I'm missing something.